src/contracts/permissions/PermissionController.sol

```diff
@@ -0,0 +1,176 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.27;
+
+import "@openzeppelin-upgrades/contracts/proxy/utils/Initializable.sol";
+import "./PermissionControllerStorage.sol";
+
+contract PermissionController is Initializable, PermissionControllerStorage {
+    using EnumerableSet for *;
+
+    modifier onlyAdmin(
+        address account
+    ) {
+        require(isAdmin(account, msg.sender), NotAdmin());
+        _;
+    }
+
+    /**
+     *
+     *                         INITIALIZING FUNCTIONS
+     *
+     */
+    constructor() {
+        _disableInitializers();
+    }
+
+    function initialize() external initializer {}
+
+    /**
+     *
+     *                         EXTERNAL FUNCTIONS
+     *
+     */
+
+    /// @inheritdoc IPermissionController
+    function setAdmin(address account, address admin) external onlyAdmin(account) {
+        // Add the admin to the account's admins
+        // If the admin is already set, the set will fail
+        EnumerableSet.AddressSet storage admins = _permissions[account].admins;
+        require(admins.add(admin), AdminAlreadySet());
+
+        emit AdminSet(account, admin);
+    }
+
+    function removeAdmin(address account, address admin) external onlyAdmin(account) {
+        EnumerableSet.AddressSet storage admins = _permissions[account].admins;
+
+        // Remove the admin from the account's admins
+        // If the admin is not set, the remove will fail
+        require(admins.remove(admin), AdminNotSet());
+
+        emit AdminRemoved(account, admin);
+    }
+
+    /// @inheritdoc IPermissionController
+    function setDelegate(
+        address account,
```

Comment:
can we update to:

delegator

delegatee

Reply:
addDelegate

Reply:
So we don't want to overload this term. How about delegate?

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/870/commits/cf26f4eb29af4a89763b2e26b99802c8fd04d090